src/recurrent/rnn-time-step.js

```diff
@@ -0,0 +1,169 @@
+import Matrix from './matrix';
+import RandomMatrix from './matrix/random-matrix';
+import Equation from './matrix/equation';
+import RNN from './rnn';
+
+export default class RNNTimeStep extends RNN {
+  constructor(options) {
+    super(options);
+  }
+
+  createInputMatrix() {
+    this.model.input = new RandomMatrix(this.inputSize, 1, 0.08);
+  }
+
+  createOutputMatrix() {
+    let model = this.model;
+    let outputSize = this.outputSize;
+    let lastHiddenSize = this.hiddenSizes[this.hiddenSizes.length - 1];
+
+    //whd
+    model.outputConnector = new RandomMatrix(outputSize, lastHiddenSize, 0.08);
+    //bd
+    model.output = new Matrix(outputSize, 1);
+  }
+
+  bindEquation() {
+    let model = this.model;
+    let hiddenSizes = this.hiddenSizes;
+    let hiddenLayers = model.hiddenLayers;
+    let equation = new Equation();
+    let outputs = [];
+    let equationConnection = model.equationConnections.length > 0
+      ? model.equationConnections[model.equationConnections.length - 1]
+      : this.initialLayerInputs
+      ;
+
+      // 0 index
+    let output = this.getEquation(equation, equation.input(model.input), equationConnection[0], hiddenLayers[0]);
+    outputs.push(output);
+    // 1+ indices
+    for (let i = 1, max = hiddenSizes.length; i < max; i++) {
+      output = this.getEquation(equation, output, equationConnection[i], hiddenLayers[i]);
+      outputs.push(output);
+    }
+
+    model.equationConnections.push(outputs);
+    equation.add(equation.multiply(model.outputConnector, output), model.output);
+    model.equations.push(equation);
+  }
+
+  /**
+   *
+   * @param {Number[]} input
+   * @returns {number}
+   */
+  runInput(input) {
+    this.runs++;
+    let model = this.model;
+    let errorSum = 0;
+    let equation;
+    while (model.equations.length < input.length - 1) {
+      this.bindEquation();
+    }
+    const outputs = [];
+
+    if (this.inputSize === 1) {
```

Comment:
Could we change this from:
```
if (...) {
  for (<loop A> ) {
    for (<loop B>) {
//      code block A
    }
  }
} else {
  for (<loop A> ) {
    for (<loop B>) {
//      code block B
    }
  }
}
```

to:

```
for (<loop A> ) {
  for (<loop B>) {
    if (...) {
//    code block A
    } else {
//    code block B
    }
  }
}
```

Man that took way longer then it should have to abstract it lol.  Again it's semicolans but I think it helps clarity (and it looks like the only line that changes is: `const error = output.weights[i] - next;` vs `const error = output.weights[i] - next[i];` unless I am missing something.

Reply:
I'm not going to waist time changing this because it is going to be deprecated.